js/ui/map.js

```diff
@@ -734,6 +734,20 @@ class Map extends Camera {
     }
 
     /**
+     * Returns a Boolean indicating whether the source is loaded.
+     *
+     * @param {string} id The ID of the source to be checked.
+     * @returns {boolean} A Boolean indicating whether the source is loaded.
+     */
+    isSourceLoaded(id) {
+        const source = this.style && this.style.sourceCaches[id];
+        if (source === undefined) {
+            throw new Error(`There is no source with ID '${id}'`);
```

Comment:
@lucaswoj ah, shoot -- this was a mistake on my part in [this comment](https://github.com/mapbox/mapbox-gl-js/pull/4033#discussion_r97206120).  I'd seen the guideline in CONTRIBUTING but didn't take it as strict/universal.  My bad.